department-of-veterans-affairs/vets-website
# Pension refactors

## Summary
On the Pension Benefits application(21P-527EZ)
- Consolidate alerts
- Update data sets to make more sense
- Update radio (key, value) select and storage patterns

### Description
By using object keys for enumeration and storage with consistent naming and casing conventions we can improve maintainability and reusability across the Pensions app.

Examples:
```
CEREMONY: 'In a civil or religious ceremony with an officiant who signed my marriage license',
OTHER: 'Some other way',
```
Rather than storing the current marriage by it's value as 'In a civil or religious ceremony with an officiant who signed my marriage license' and 'Some other way', we store it by it's key as 'CEREMONY' or 'OTHER'.

```
MARRIED: 'Married',
NEVER_MARRIED: 'Never Married',
SEPARATED: 'Separated',
WIDOWED: 'Widowed',
DIVORCED: 'Divorced',
```
Rather than storing the martial status by it's value 'Never married', we store it as  'NEVER_MARRIED'  and vice-versa.
````
  VETERAN: 'Veteran',
  SPOUSE: 'Veteran’s spouse',
  DEPENDENT: 'Veteran’s child',
````
Rather than storing recipient and receiver as separate key, values, we will use the same key, value pairs.


## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_form_enabled` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_form_enabled
3. Run `vets-website` and `vets-api`

## How to verify
1. Go to `http://localhost:3001/pension/application/527EZ` in your browser
2. Start or continue the application
3. Verify the application is able to be saved and submitted

## Related issue(s)
[Epic in Github](https://app.zenhub.com/workspaces/benefits-pension-64e775aaa6b1ca1ed49b2ede/issues/zh/224)

## Testing done


## What areas of the site does it impact?
Pension Benefits Application

## Acceptance criteria
[Ticket in Github](https://app.zenhub.com/workspaces/benefits-pension-64e775aaa6b1ca1ed49b2ede/issues/zh/224)

### Quality Assurance & Testing

### Error Handling


### Authentication
